traits/editor_factories.py

```diff
@@ -143,12 +143,17 @@ def _expects_hastraits_instance(handler):
 def _instance_handler_factory(handler):
     """ Get the instance factory of an Instance or TraitInstance
     """
-    from traits.api import BaseInstance, TraitInstance
+    from traits.api import BaseInstance, DefaultValue, TraitInstance
 
     if isinstance(handler, TraitInstance):
         return handler.aClass
     elif isinstance(handler, BaseInstance):
-        return handler.default_value
+        result = handler.default_value
```

Comment:
Thanks for the change. At this point there's no need to compute `result` up front: the code can be simplified to inline the `result` variable.

Reply:
Ah, sorry; I missed that you'd already merged. Never mind ...

Reply:
Ah my mistake, I should have waited for a confirmation after the last change.  Im happy to do this clean up in a short follow up PR